xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-default/src/main/java/org/xwiki/notifications/internal/NotificationDisplayerComponent.java

```diff
@@ -0,0 +1,185 @@
+     * @param templateManager the {@link TemplateManager} to use
+     * @param scriptContextManager the {@link ScriptContextManager} to use
+     * @param componentManager the {@link ComponentManager} to use
+     * @param baseObject the XObject which has the required properties to instantiate the component
+     * @throws NotificationException if the properties of the given BaseObject could not be loaded
+     */
+    public NotificationDisplayerComponent(BaseObjectReference objectReference, DocumentReference authorReference,
+            TemplateManager templateManager, ScriptContextManager scriptContextManager,
+            ComponentManager componentManager, BaseObject baseObject) throws NotificationException
+    {
+        this.objectReference = objectReference;
+        this.authorReference = authorReference;
+        this.templateManager = templateManager;
+        this.scriptContextManager = scriptContextManager;
+        this.componentManager = componentManager;
+        this.setProperties(baseObject);
+    }
+
+    /**
+     * Set the object attributes by extracting their values from the given BaseObject.
+     *
+     * @param baseObject the XObject that should contain the parameters
+     * @throws NotificationException if an error occured while extracting the parameters from the base object
+     */
+    private void setProperties(BaseObject baseObject) throws NotificationException
+    {
+        try {
+            this.notificationTemplate = baseObject.getStringValue(
+                    NotificationDisplayerDocumentInitializer.NOTIFICATION_TEMPLATE);
+            this.eventType = baseObject.getStringValue(NotificationDisplayerDocumentInitializer.EVENT_TYPE);
+        } catch (Exception e) {
+            throw new NotificationException(
+                    String.format("Unable to extract the parameters from the [%s] NotificationDisplayerClass.",
+                            baseObject), e);
+        }
+    }
+
+    @Override
+    public Block renderNotification(CompositeEvent eventNotification) throws NotificationException
+    {
+        try {
+            // Allow the template to access the event during its execution
+            scriptContextManager.getCurrentScriptContext().setAttribute(EVENT_BINDING_NAME, eventNotification,
+                    ScriptContext.ENGINE_SCOPE);
+
+            // If we have no template defined, fallback on the default displayer
+            if (StringUtils.isBlank(this.notificationTemplate)) {
+                return ((NotificationDisplayer) this.componentManager.getInstance(NotificationDisplayer.class))
+                        .renderNotification(eventNotification);
+            }
+
+            // Render the template
+            Template customTemplate = templateManager.createStringTemplate(
+                    this.notificationTemplate,
+                    this.getAuthorReference());
+
+            return templateManager.getXDOM(customTemplate);
```

Comment:
`getXDOM()` only parse the template but does not execute it (except for the special use case where no source syntax is provided), you should use `execute()` instead.